src/ansible_navigator/configuration_subsystem/utils.py

```diff
@@ -0,0 +1,174 @@
+"""Utilities related to the configuration subsystem."""
+
+import sys
+
+from types import SimpleNamespace
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Union
+
+from .definitions import ApplicationConfiguration
+from .definitions import CliParameters
+from .definitions import Constants as C
+from .definitions import SettingsEntry
+
+
+class HumanReadableEntry(NamedTuple):
+    """Data structure for a setting entry."""
+
+    #: The possible values
+    choices: List
+    #: The CLI parameters, short and long
+    cli_parameters: Dict[str, str]
+    #: The path to the current settings file
+    current_settings_file: str
+    #: The current, effective value
+    current_value: Union[bool, Dict, str, List]
+    #: The default value
+    default: Union[bool, str]
+    #: A short description
+    description: str
+    #: The environment variable
+    env_var: str
+    #: Indicate if the current == the default
+    is_default: bool
+    #: The name
+    name: str
+    #: A sample settings file snippet
+    settings_file_sample: Union[str, Dict]
+    #: The source of the current value
+    source: str
+
+
+def transform_settings(
+    settings: ApplicationConfiguration,
+) -> List[Dict[str, Union[bool, Dict, str, List]]]:
+    """Transform the current settings into a list of dictionaries.
+
+    :param settings: The current settings
+    :returns: The settings represented as a list of dictionaries
+    """
+    settings_list = []
+
+    entry = _settings_file_entry(settings.internals)
+    settings_list.append(entry._asdict())
+
+    for current in settings.entries:
+        application_name = settings.application_name
+        settings_file_path = settings.internals.settings_file_path
+        entry = _standard_entry(
+            current=current,
+            application_name=application_name,
+            settings_file_path=settings_file_path,
+        )
+        # py36, py37 dataclass._asdict returns OrderedDict
+        if sys.version_info >= (3, 8):
+            entry_as_dict = entry._asdict()
+        else:
+            entry_as_dict = dict(entry._asdict())
+        settings_list.append(entry_as_dict)
+
+    sorted_settings = sorted(settings_list, key=lambda d: d["name"])
```

Comment:
FYI, the `operator` module has a helper for this case already:
```python
import operator
```
```suggestion
    sorted_settings = sorted(settings_list, key=operator.itemgetter("name"))
```

Reply:
so when I first was making a pass at this, I attempted to use the operator method `itemgetter` but it was throwing an error on something around a str not being callable. Might have been pore syntax from me but this looks close to what I had implemented. happy to change it back and get it working but do we want another import since lambda is builtin?

Reply:
Up to you.

Reply:
refactored to not use either